### PR TITLE
Detect invalid characters in text and set a risk. Fixes #1347.

### LINKED ISSUE #1347: Detect invalid chars in a DNS query and set risk
After an internal discussion we have decided that is should be nDPI to check possibly invalid chars in a DNS query, and not ntopng.

For reference see:
https://github.com/ntop/ntopng/blob/64252fbcea496faad44dc296a0f8ce8964083cb9/src/Flow.cpp#L786

So basically when invalid chars are found:
- A risk should be set
- A clean copy of the DNS query should be available, e.g., either `ndpiFlow->host_server_name` or in another field.

### PR DESCRIPTION
**Pull request by utoni**
Signed-off-by: Toni Uhlig <matzeton@googlemail.com>

### CODE REVIEW

File: src/include/ndpi_typedefs.h [hunk 1]
```diff
@@ -114,6 +114,7 @@ typedef enum {
   NDPI_CLEAR_TEXT_CREDENTIALS,
   NDPI_DNS_LARGE_PACKET,
   NDPI_DNS_FRAGMENTED,
+  NDPI_INVALID_CHARACTERS,
```
Comment (IvanNardi) [inline src/include/ndpi_typedefs.h:117]: Could you update also wireshark/ndpi.lua and doc/flow_risks.rst, please?

Comment (utoni) [reply src/include/ndpi_typedefs.h:117]: Thanks, I've forgot that we have a documentation now. Awesome!

File: wireshark/ndpi.lua [hunk 2]
```diff
@@ -77,6 +77,7 @@ flow_risks[35] = ProtoField.bool("ndpi.flow_risk.suspicious_entropy", "Suspiciou
 flow_risks[36] = ProtoField.bool("ndpi.flow_risk.clear_text_credentials", "Cleat-Text credentials", num_bits_flow_risks, nil, bit(4), "nDPI Flow Risk: cleat-text credentials")
 flow_risks[37] = ProtoField.bool("ndpi.flow_risk.dns_large_packet", "DNS large packet", num_bits_flow_risks, nil, bit(5), "nDPI Flow Risk: DNS packet is larger than 512 bytes")
 flow_risks[38] = ProtoField.bool("ndpi.flow_risk.dns_fragmented", "DNS fragmented", num_bits_flow_risks, nil, bit(6), "nDPI Flow Risk: DNS message is fragmented")
+flow_risks[39] = ProtoField.bool("ndpi.flow_risk.invalid_characters", "Invalid characters", num_bits_flow_risks, nil, bit(7), "nDPI Flow Risk: Text contains non-printable characters")
 
 -- Last one: keep in sync the bitmask when adding new risks!!
 flow_risks[64] = ProtoField.new("Unused", "ndpi.flow_risk.unused", ftypes.UINT32, nil, base.HEX, bit(32) - bit(7))
```
Comment (IvanNardi) [inline wireshark/ndpi.lua]: \- bit(8))

**Review by IvanNardi — APPROVED** [review]
Small issue in lua code.
Other than that, everything seems fine!

**Pull request merged**